### PR TITLE
Misc 3.4 Backports

### PR DESCRIPTION
**Pull request by nyalldawson**
Backporting some misc fixes

### CODE REVIEW

File: src/app/qgsapplayertreeviewmenuprovider.cpp [hunk 1]
```diff
@@ -691,6 +691,8 @@ void QgsAppLayerTreeViewMenuProvider::editVectorSymbol()
     singleRenderer->setSymbol( symbol.release() );
     layer->triggerRepaint();
     mView->refreshLayerSymbology( layer->id() );
+    layer->emitStyleChanged();
+    QgsProject::instance()->setDirty( true );
```
Comment (m-kuhn) [inline src/app/qgsapplayertreeviewmenuprovider.cpp]: Would it make sense to connect the layer style changed signal directly to a project setDirty slot (by the project, when a layer is added)?

Comment (nyalldawson) [reply src/app/qgsapplayertreeviewmenuprovider.cpp]: That signal is fired for internal reasons too, whereas dirtying should only ever happen as a direct result of user interaction

Comment (m-kuhn) [reply src/app/qgsapplayertreeviewmenuprovider.cpp]: In this case a `void emitStyleChanged( bool makeProjectDirty = true );` might be something to consider.

Just thinking it's too easy to miss this.

Comment (nyalldawson) [reply src/app/qgsapplayertreeviewmenuprovider.cpp]: Well really having an emitXxx method in the first place is an antipattern, hinting that this whole part of the api needs a rework.

Comment (m-kuhn) [reply src/app/qgsapplayertreeviewmenuprovider.cpp]: Ok, thanks!

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**